Makefile.qemu.inc

```diff
@@ -4,6 +4,7 @@
 MONITOR ?= "/dev/null"
 
 run: $(IMAGE)
```

Comment:
Hmmm. I know this technically works, but seems odd. It only works because Makefile.qemu.inc and Makefile.mainboard.inc happen to both be included by every board in  `src/mainboard/emulation`.

Also the mainboard target is just `mainboard: $(IMAGE)` so the run target depending `$(IMAGE)` should already do the trick.

Reply:
So the is off the back of the Makefile changes that went in today, which appear to have broken `make run` triggering a rebuild of the mainboard if there were changes therein. On the call this was the suggested workaround, but one thing I can say for sure is that it wasn't working for me and Titus as-is.

Reply:
out of curiosity, does `run: $(IMAGE) mainboard` fix the problem, or no?

Reply:
Looks like that works for me. So as a small cleanup on this PR I'd prefer that to explicitly calling `make mainboard`

Reply:
Oh another think to try. In the emulation/*/Makefile files, switching the include order seems to also work. Including mainboard.inc before qemu.inc.

Reply:
Yep yeah agree - just bad at `make`. Fixed!